automates/text_reading/src/main/scala/org/clulab/aske/automates/OdinActions.scala

```diff
@@ -163,36 +210,104 @@ class OdinActions(val taxonomy: Taxonomy, expansionHandler: Option[ExpansionHand
+    val mns = new ArrayBuffer[Mention]()
+
+    // group by sentence
+    val sentGroup = mentions.filter(_.arguments.contains("variable")).groupBy(_.sentence)
+    for ((sentId, sameSentMentions) <- sentGroup) {
+      // group by group
+      val spanGroups = sameSentMentions.groupBy(_.tokenInterval)
+
+      for (sg <- spanGroups) {
+        // check the max number of args (conj defs type 2 have at least two var-def pairs (i.e., at least 4 args)
+        val maxNumOfArgs = sg._2.maxBy(_.arguments.values.flatten.toList.length).arguments.values.flatten.toList.length
+        // check the max num of variables in the mentions in the overlapping group - we want to preserve conj defs and those will have most vars
+        val maxNumOfVars = sg._2.maxBy(_.arguments("variable").length).arguments("variable").length
+        // chose a mention with most args and most vars - if they have the same span and same (max) num of args and vars, it shouldnt matter which one it is, so take the first one
+        val chosenMen = sg._2.filter(m => m.arguments("variable").length == maxNumOfVars & m.arguments.values.flatten.toList.length==maxNumOfArgs).head
+        mns.append(chosenMen)
+      }
+    }
+
     val mens = mns.toList
     mens.toVector.distinct
   }
 
+  def filterDefsByOffsets(mentions: Seq[Mention], filterBy: String ,state: State = new State()): Seq[Mention] = {
+    // get rid of overlapping definitions; depending on when we need to use it, we will check def start offset or end offset - it also makes sense to do both directions
+    val mns = new ArrayBuffer[Mention]()
+
+    // group by sentence
+    val sentGroup = mentions.filter(_.arguments.contains("variable")).groupBy(_.sentence)
+    for ((sentId, sameSentMentions) <- sentGroup) {
+
+      val spanGroups = filterBy match {
+        case "varAndDefStartOffset" => sameSentMentions.filter(_.arguments.contains("variable")).groupBy(m => (m.arguments("variable").head, m.arguments("definition").head.startOffset))
+        case "varAndDefEndOffset" => sameSentMentions.filter(_.arguments.contains("variable")).groupBy(m => (m.arguments("variable").head, m.arguments("definition").head.endOffset))
+        case _ => ???
+
+      }
+
+      for (sg <- spanGroups) {
+        // out of the overlapping group, choose the one with longest def
+        val chosenMen = sg._2.maxBy(_.arguments("definition").head.text.length)
+        mns.append(chosenMen)
+      }
+    }
+
+    mns.toVector.distinct
+  }
+
+  def noOverlapInGivenArg(mention: Mention, argType: String): Boolean = {
+    // check if mention contains overlapping args of a given type
+    val argsOfGivenType = mention.arguments(argType)
+    val groupedByTokenInt = groupByTokenOverlap(argsOfGivenType)
+    // if there is an overlap in args, those will be grouped => the number of groups will be lower than the number of args
+    if (groupedByTokenInt.keys.toList.length == argsOfGivenType.length) true else false
```

Comment:
@maxaalexeeva the following code snippet:
```python
if <condition>:
    return True
else:
    return False
```
... can always be simplified to:
```python
return <condition>
```

I think this line follows that pattern. Can you please do the simplification?

Reply:
@pauldhein, right! Thanks! Pushed! *can even not say 'return'!